File: java/primes.java
public class Primes {
    public static void main(String[] args) {
        int limit = 100000000;
        for (int i = 2; i <= limit; i++) {
            boolean isPrime = true;
            for (int j = 2; j * j <= i; j++) {
                if (i % j == 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                System.out.println(i);
            }
        }
    }
}

File: zig/primes.zig
const std = @import("std");
const warn = std.debug.warn;

pub fn printPrimes(limit: usize) void {
    var primes: [limit + 1]bool = undefined;
    primes[0] = false;
    primes[1] = false;

    for (i: usize = 2; i <= limit; i += 1) : (primes[i] = true) {}

    for (i: usize = 2; i * i <= limit; i += 1) {
        if (primes[i]) {
            for (j: usize = i * i; j <= limit; j += i) : (primes[j] = false) {}
        }
    }

    for (i: usize = 2; i <= limit; i += 1) {
        if (primes[i]) {
            warn("{}\n", .{i});
        }
    }
}

pub fn main() !void {
    printPrimes(1000000);
    return .Ok;
}
